### PR TITLE
Add http -> https redirect

### PR DESCRIPTION
**Pull request by joepurnell1**
**JIRA:**

**Breakdown:**
- Add a redirect from http port 80 to https of the same url
- Add ingress rule for http port 80 into the security group

### CODE REVIEW

File: lb-https.tf [hunk 1]
```diff
@@ -20,6 +24,22 @@ resource "aws_alb_listener" "https" {
...
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
```
Comment (kelemensanyi) [inline lb-https.tf]: Move to lb-http.tf

File: lb-https.tf [hunk 2]
```diff
@@ -6,6 +6,10 @@ variable "https_port" {
   default = "443"
 }
 
+variable "http_port" {
+  default = "80"
+}
+
```
Comment (kelemensanyi) [inline lb-https.tf]: I think we should create a separate file for the HTTP specific config, and name it lb-http.tf.
Basically all the new configuration related to HTTP -> HTTPS redirect should go there.

File: lb-https.tf [hunk 3]
```diff
@@ -30,3 +50,13 @@ resource "aws_security_group_rule" "ingress_lb_https" {
...
+resource "aws_security_group_rule" "ingress_lb_http" {
+  type              = "ingress"
+  description       = "HTTP"
+  from_port         = var.http_port
+  to_port           = var.http_port
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.nsg_lb.id
+}
+
```
Comment (kelemensanyi) [inline lb-https.tf]: Move to lb-http.tf

File: lb-https.tf (hunk 1 above)
Comment (joepurnell1) [reply lb-https.tf]: Done

File: lb-https.tf (hunk 2 above)
Comment (joepurnell1) [reply lb-https.tf]: Done

File: lb-https.tf (hunk 3 above)
Comment (joepurnell1) [reply lb-https.tf]: Done

File: lb-http.tf [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+variable "http_port" {
+  default = "80"
+}
+
+variable "certificate_arn" {}
```
[the remark this answers is not in the document]
Comment (joepurnell1) [reply lb-http.tf]: done

**Pull request merged**